inc/Logger.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "ILogOutputter.hpp"
+#include "LogData.hpp"
+
+namespace weblog {
+
+/**
+ * @brief Logger class to log messages.
+ *
+ * Implements the Singleton pattern to log messages to different outputters.
+ * The Logger class is a Singleton class, so it can be accessed from anywhere in the code.
+ * The Logger class can log messages to different outputters, which implement the ILogOutputter interface.
+ * The Logger class can log messages with different log levels, which are defined in the LogLevel enum.
+ * The Logger class can be initialized with a log level and an outputter.
+ *
+ */
+class Logger {
+
+public:
+	static Logger& getInstance();
+	Logger& addLogOutputter(ILogOutputter* outputter);
+
+	void operator+=(const LogData& logData) const;
+
+	static Logger& init(LogLevel level = LevelInfo, ILogOutputter* outputter = NULL);
+	LogLevel getLevel() const;
+	Logger& setLevel(LogLevel level);
+
+private:
+	// private constructor to prevent direct object creation
+	Logger();
+
+	// disallow copying the instance
+	Logger(const Logger& ref);
```

Comment:
There is: we have a singleton, so we don't want anyone to be able to construct multiple instances of the thing. Since I can't use delete keyword I need to make them private

Reply:
yeah, I realised now, see above.